### PR TITLE
[openshift][workflow][Makefile][HCAP-#585] Deploy config changes

### PR DESCRIPTION
**Pull request by fwpushan**
----
1. Create make alias for testing deploy config change
2. Create make alias for process and apply deployment config change
3. Create new github workflow config.

Ticket: [HCAP-585](https://freshworks.atlassian.net/browse/HCAP-585)

- [x] All secrets configured.
- [x] Passing all checks

### CODE REVIEW

**Review by Zenen-freshworks — DISMISSED** [review]

**Review by will-freshworks — CHANGES_REQUESTED** [review]
The code looks good, but I am wondering why we have a single `config` tag for pushing to all three environments? Seems risky. Why is the config not bundled along with the rest of the `prod`/`test`/`dev` code?

I am probably missing something and need more context.

**Comment by fwpushan** [comment]
> The code looks good, but I am wondering why we have a single `config` tag for pushing to all three environments? Seems risky. Why is the config not bundled along with the rest of the `prod`/`test`/`dev` code?
> 
> I am probably missing something and need more context.

Initially. I thought about putting config update as part of dev/test/prod general deployment.
But as per current process, team separate config change from build promotion.
Currently, config change is manual process, dev manually config the env before promote.
Deployment has image trigger (unlike Court), so tagging image automatically start deployment to respective env
So, introducing config update in same  flow will trigger rollout in respective env twice (redundant) one for config change and one for image tagging.
For above reason, I also kept it separate from general image promotion.
Now, what is your recommendation?

File: .github/workflows/config.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      OS_NAMESPACE_SUFFIX: test
+    needs: test
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Apply changes
+        run: |
+          oc login --token="$AUTH_TOKEN" --server="$CLUSTER"
+          cd "$GITHUB_WORKSPACE"
+          make server-config
```
Comment (vyasworks) [inline .github/workflows/config.yml:66]: Is reapplying `server-config` going to trigger a deployment config? I thought DC resource would change in K8S and wait for the actual trigger configured in DC or forced via `server-deploy`  my understanding was different. I might be wrong :)

Comment (fwpushan) [reply .github/workflows/config.yml:70]: Now using same tag to update config. Please review

**Review by Zenen-freshworks — DISMISSED** [review]
looks good to me

**Review by dbayly-freshworks — APPROVED** [review]

File: .github/workflows/config-prod.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Test and Deploy Config on Prod
+
+on:
+  push:
+    paths:
+      - 'openshift/*.yml'
```
Comment (will-freshworks) [inline .github/workflows/config-prod.yml:6]: I think `openshift/**` is safer

**Review by will-freshworks — CHANGES_REQUESTED** [review]
Needs the change we discussed so dev and test can be deployed separately.

**Review by will-freshworks — APPROVED** [review]

**Review by Zenen-freshworks — APPROVED** [review]

**Review by dbayly-freshworks — APPROVED** [review]

**Pull request merged**